File: Property_management_system/src/com/property/management/MaintenanceRequest.java
package com.property.management;

import java.util.ArrayList;
import java.util.List;

public class MaintenanceRequest {
    private String requestID;
    private String tenantID;
    private String propertyID;
    private String description;
    private String status;
    private String dateSubmitted;
    public static List<MaintenanceRequest> requests = new ArrayList<>();

    public MaintenanceRequest(String requestID, String tenantID, String propertyID, String description, String status, String dateSubmitted) {
        this.requestID = requestID;
        this.tenantID = tenantID;
        this.propertyID = propertyID;
        this.description = description;
        this.status = status;
        this.dateSubmitted = dateSubmitted;
    }

    // Getters and Setters
    public String getRequestID() {
        return requestID;
    }

    public void setRequestID(String requestID) {
        this.requestID = requestID;
    }

    public String getTenantID() {
        return tenantID;
    }

    public void setTenantID(String tenantID) {
        this.tenantID = tenantID;
    }

    public String getPropertyID() {
        return propertyID;
    }

    public void setPropertyID(String propertyID) {
        this.propertyID = propertyID;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getDateSubmitted() {
        return dateSubmitted;
    }

    public void setDateSubmitted(String dateSubmitted) {
        this.dateSubmitted = dateSubmitted;
    }

    // Methods
    public static void submitRequest(MaintenanceRequest request) {
        requests.add(request);
        System.out.println("Maintenance request submitted: " + request.requestID);
    }

    public static void trackRequestStatus(String requestID) {
        for (MaintenanceRequest request : requests) {
            if (request.requestID.equals(requestID)) {
                System.out.println("Request status: " + request.status);
                return;
            }
        }
        System.out.println("Request not found!");
    }
}

File: Property_Management_Systems/src/maintenance_request.py
class MaintenanceRequest:
    requests = []

    def __init__(self, request_id, tenant_id, property_id, description, status, date_submitted):
        self.request_id = request_id
        self.tenant_id = tenant_id
        self.property_id = property_id
        self.description = description
        self.status = status
        self.date_submitted = date_submitted

    @classmethod
    def submit_request(cls, request):
        cls.requests.append(request)
        print(f"Maintenance request {request.request_id} submitted.")

    @classmethod
    def track_request_status(cls, request_id):
        for request in cls.requests:
            if request.request_id == request_id:
                print(f"Request status: {request.status}")
                return
        print("Request not found!")

    @classmethod
    def view_requests(cls):
        for request in cls.requests:
            print(f"ID: {request.request_id}, Tenant ID: {request.tenant_id}, Property ID: {request.property_id}, Description: {request.description}, Status: {request.status}, Date: {request.date_submitted}")

 # Input
request1 = MaintenanceRequest(
    "R001", "T001", "P001", "Leaky faucet", "Pending", "2024-07-25")
MaintenanceRequest.submit_request(request1)
MaintenanceRequest.view_requests()

MaintenanceRequest.track_request_status("R001")
